File: vm/home/tango-cs/training/python-api/reading01.py
#/!/usr/bin/python
#
# client example: reading an attribute , also with simplified syntax
# C. Scafuri - jan. 2016
#

from PyTango import *
import sys

devname = 'sys/tg_test/1'
attname = 'double_scalar'

#step 1) create the device proxy
try:
    dev = DeviceProxy(devname)
    print 'proxy for %s created' %(devname)
except DevFailed, ex:
        Except.print_exception(ex)
        sys.exit(1) #semplicistic error handling
        
#step 2 read the attribute
try:
    da = dev.read_attribute(attname) #read the attribute
    value=da.value
except DevFailed, ex:
    Except.print_exception(ex)
    print 'failed to read attribute'
    sys.exit(1) #simplicistic error handling

print '%s/%s   value: %f ' % (devname,attname,value)

#step3 use simplified interface

try:
    value2 = dev.double_scalar
except DevFailed, ex:
    Except.print_exception(ex)
    print 'failed to read double_scalar'
    sys.exit(1) #simplicistic error handling

    
print '%s/%s  value2: %f ' % (devname,attname,value2)
sys.exit(0)

File: vm/home/tango-cs/training/client-api/reading01.cpp
/*
 * reading01.cpp
 * 
 *  example program for Tango training - reading an attribute
 * 
 * C. Scafuri
 * Elettra - Sincrotrone Trieste S.C.p.A.
 * 2016
 */



#include <tango.h>
#include <cstdlib>
using namespace Tango;
int main(int argc, char **argv)
{
    string devname = "sys/tg_test/1";
    string attname = "double_scalar";
    DeviceProxy *dev = 0;
    // step 1) create the device proxy
    try{
        dev = new Tango::DeviceProxy(devname);
    }
    catch(Tango::DevFailed &e)
    {
        Tango::Except::print_exception(e);
        cerr << "failed to create DeviceProxy" << endl;
        exit(EXIT_FAILURE); // semplicistic error handling

    }
    
    // prepare the reading
    DeviceAttribute da;
    double value=0.0;
    
    try{
        da = dev->read_attribute(attname); //read the attribute
        da >> value;  //use overoladed operator to extract the value
    }
    catch(DevFailed &e)
    {
        Tango::Except::print_exception(e);
        cerr << "failed to read attribute" << endl;
        delete dev; // free resources
        exit(EXIT_FAILURE); // semplicistic error handling
    }
    
    cout << devname << "/" << attname << " value: "<< value << endl;
    delete dev;
    exit(EXIT_SUCCESS);
}